src/uri/UriValidator.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <uprotocol-cpp/uri/validator/UriValidator.h>
+#include "uprotocol/uri.pb.h"
+#include "uprotocol/ustatus.pb.h"
+#include <uprotocol-cpp/uri/serializer/LongUriSerializer.h>
+#include <uprotocol-cpp/uri/serializer/MicroUriSerializer.h>
+#include <string.h>
+
+using namespace uprotocol::uri;
+using namespace uprotocol::v1;
+
+/**
+ * isValidUri a {@link uri} to ensure that it has at least a name for the
```

Comment:
Add @brief for all brief method descriptions.

Reply:
Done